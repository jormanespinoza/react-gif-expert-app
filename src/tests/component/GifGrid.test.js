import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Testing GifGrid component', () => {

    const category = 'Death Note';

    test('should render <GifGrid /> properly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should show items once images al loader by useFetchGifs', () => {
        const gifs = [
            {
                id: '1',
                url: 'test-url-1',
                title: 'test-title'
            },
            {
                id: '2',
                url: 'test-url-2',
                title: 'test-title-2'
            }
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});
