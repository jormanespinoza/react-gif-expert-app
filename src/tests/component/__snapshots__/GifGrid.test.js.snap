// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid component should render <GifGrid /> properly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Death Note
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
